modules/margin-protocol/src/lib.rs

```diff
@@ -350,28 +350,26 @@ impl<T: Trait> Module<T> {
 			.ok_or(Error::<T>::NumOutOfBound)?;
 
 		// realizing
-		let balance_delta_abs = u128_from_fixed_128(balance_delta.saturating_abs());
 		if balance_delta.is_positive() {
-			// trader has profit
-			let realized = cmp::min(
-				<T::LiquidityPools as LiquidityPools<T::AccountId>>::liquidity(position.pool),
-				balance_delta_abs,
-			);
+			// trader has profit, max realizable is the pool's liquidity
+			let pool_liquidity = fixed_128_from_u128(<T::LiquidityPools as LiquidityPools<T::AccountId>>::liquidity(
+				position.pool,
+			));
+			let realized = cmp::min(pool_liquidity, balance_delta);
 			<T::LiquidityPools as LiquidityPools<T::AccountId>>::withdraw_liquidity(
 				&Self::account_id(),
 				position.pool,
-				realized,
+				u128_from_fixed_128(realized),
 			)?;
-			<Balances<T>>::mutate(who, |b| *b += realized);
+			Self::_update_balance(who, realized);
 		} else {
 			// trader has loss
-			let realized = cmp::min(Self::balances(who), balance_delta_abs);
 			<T::LiquidityPools as LiquidityPools<T::AccountId>>::deposit_liquidity(
 				&Self::account_id(),
 				position.pool,
-				realized,
+				u128_from_fixed_128(balance_delta.saturating_abs()),
```

Comment:
What happens if the margin protocol doesn't have enough funds to pay the `balance_delta` amount to the liquidity pool?